From 8b668989bc17433ab507ec442b51a1518b593604 Mon Sep 17 00:00:00 2001
From: Awais Belal <awais_belal@mentor.com>
Date: Thu, 9 Feb 2017 18:15:45 +0500
Subject: [PATCH] CrossCompile.cmake: strip sysroot info from build variables

This is to align with the recipe specific sysroot mechanism
which will otherwise break the build for any other package
that utilizes LLVM by specifying the LLVM sysroot on the
compiler/linker commandline rather than that package's
specific sysroot.

Signed-off-by: Awais Belal <awais_belal@mentor.com>
---
 cmake/modules/CrossCompile.cmake | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/cmake/modules/CrossCompile.cmake b/cmake/modules/CrossCompile.cmake
index 6378b7e..9682a98 100644
--- a/cmake/modules/CrossCompile.cmake
+++ b/cmake/modules/CrossCompile.cmake
@@ -36,7 +36,9 @@ function(llvm_create_cross_target_internal target_name toolchain buildtype)
     WORKING_DIRECTORY ${LLVM_${target_name}_BUILD}
     DEPENDS ${LLVM_${target_name}_BUILD}
     COMMENT "Configuring ${target_name} LLVM..."
-    COMMAND "${CMAKE_COMMAND}" "-E" "copy" "${CMAKE_SOURCE_DIR}/../build/tools/llvm-config/BuildVariables.inc" "${CMAKE_SOURCE_DIR}/../build/NATIVE/tools/llvm-config/BuildVariables.inc")
+    COMMAND "${CMAKE_COMMAND}" "-E" "copy" "${CMAKE_SOURCE_DIR}/../build/tools/llvm-config/BuildVariables.inc" "${CMAKE_SOURCE_DIR}/../build/NATIVE/tools/llvm-config/BuildVariables.inc"
+    COMMAND sed -i "'s/\\(.*--sysroot=\\\)[^ ]* \\(.*\\)/\\1 \\2/g'" "${CMAKE_SOURCE_DIR}/../build/NATIVE/tools/llvm-config/BuildVariables.inc"
+    COMMAND sed -i "'s/--sysroot=//g'" "${CMAKE_SOURCE_DIR}/../build/NATIVE/tools/llvm-config/BuildVariables.inc")
 
   add_custom_target(CONFIGURE_LLVM_${target_name}
                     DEPENDS ${LLVM_${target_name}_BUILD}/CMakeCache.txt)
-- 
1.9.1

